### PR TITLE
Feat: Create new Experimental setting under Test Options settings of OONI Probe apps

### PR DESCRIPTION
**Pull request by aanorbel**
Fixes https://github.com/ooni/probe/issues/2266

## Proposed Changes

  - Add `Experimental` under `Test Options` settings.
  - Add `long_running_tests_in_foreground` preference.
      - Ensure `torsf` and `vanilla_tor` are annotated properly and run based on preference.
  - Ensure all tests are added to automated runs.
      - Ensure `dash` and `ndt` under `performance` test suite are not run in automated runs 
  - Add copy in Test Options screen to mention that test settings apply to both manual and automated runs.

|.|.|
|-|-|
| Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 11 58 25 | Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 11 58 32 |
| Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 11 58 35 | Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 11 59 00 |
| Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 12 01 26 | Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 12 01 33 |
| Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 12 01 40 | Simulator Screen Shot - iPhone 14 Pro Max - 2023-02-03 at 12 01 45 |

### CODE REVIEW

File: ooniprobe/Test/Suite/PerformanceSuite.m [hunk 1]
```diff
@@ -18,9 +18,9 @@ - (NSArray*)getTestList {
             [self.testList addObject:[[NdtTest alloc] init]];
         if ([SettingsUtility getSettingWithName:@"run_dash"])
             [self.testList addObject:[[Dash alloc] init]];
-        if ([SettingsUtility getSettingWithName:@"run_http_invalid_request_line"])
+        if ([SettingsUtility getSettingWithName:@"run_http_invalid_request_line"] && !self.autoRun)
```
Comment (bassosimone) [inline ooniprobe/Test/Suite/PerformanceSuite.m:21]: Why?

File: ooniprobe/Test/Suite/PerformanceSuite.m [hunk 2]
```diff
@@ -18,9 +18,9 @@ - (NSArray*)getTestList {
...
             [self.testList addObject:[[HttpInvalidRequestLine alloc] init]];
-        if ([SettingsUtility getSettingWithName:@"run_http_header_field_manipulation"])
+        if ([SettingsUtility getSettingWithName:@"run_http_header_field_manipulation"] && !self.autoRun)
```
Comment (bassosimone) [inline ooniprobe/Test/Suite/PerformanceSuite.m:23]: Why?

**Review by bassosimone — COMMENTED** [review]
I need to better understand the changes at `ooniprobe/Test/Suite/PerformanceSuite.m`, otherwise this looks good to me! 🙌

**Review by bassosimone — COMMENTED** [review]
The implementation LGTM 🐳 

However, when I tested this diff something unexpected happened:

1. I ran the experimental suite only and it did not ran long running tests

2. I ran all tests and it ran the long running tests

3. I ran the experimental suite again and it ran the long running tests

Can you double check this, @aanorbel?

**Comment by aanorbel** [comment]
> Can you double check this, @aanorbel?

Actual problem description,

- [`getTestList`](https://github.com/ooni/probe-ios/blob/issues/2266/ooniprobe/Test/Suite/ExperimentalSuite.m#L15) prevents the items in `testList` from changing after loaded on app startup.
- Restarting the app with change in settings (`long_running_tests_in_foreground`) gives the correct behaviour.

I will proceed to correct this @bassosimone .

**Comment by bassosimone** [comment]
Thank you, @aanorbel !

**Review by bassosimone — APPROVED** [review]
🐳

**Pull request merged**